### ISSUE TITLE
Unhandled exception (TaskScheduler.UnobservedTaskException)

### ISSUE THREAD

**Issue by Dubickdsa** [RatScanner/RatScanner#375]
**Error**
Unhandled exception (TaskScheduler.UnobservedTaskException)
```
System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (Value cannot be null. (Parameter 'value'))
 ---> System.ArgumentNullException: Value cannot be null. (Parameter 'value')
   at Newtonsoft.Json.Utilities.ValidationUtils.ArgumentNotNull(Object value, String parameterName)
   at Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings)
   at RatTracking.ProgressDB.Init() in D:\Programmierung\Projects\RatScanner\RatScanner\RatTracking\ProgressDB.cs:line 23
   at RatScanner.RatScannerMain..ctor()
   at RatScanner.RatScannerMain.get_Instance()
   at RatScanner.View.BlazorUI.<>c.<.ctor>b__8_0(IServiceProvider s)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.D
```
<details>
<summary>Log</summary>

```
[21:12:56.2425510] > [Info]  ----- RatScanner 3.2.3 -----
[21:12:56.2427298] > [Info]  Starting RatScanner...
[21:12:56.2427327] > [Info]  Checking for updates...
[21:12:56.2445748] > [Info]  Loading resource "RSClientVersion"...
[21:12:56.7845060] > [Info]  Loading config...
[21:12:56.7853171] > [Info]  Checking for item data updates...
[21:12:56.7857022] > [Info]  Loading resource "RSItemDataBundleVersion"...
[21:12:56.9273757] > [Info]  Loading progress data...
[21:14:37.2283138] > [Error] Unhandled exception (Application.Current.DispatcherUnhandledException)
 -------------------- 
 System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.ArgumentNullException: Value cannot be null. (Parameter 'value')
   at Newtonsoft.Json.Utilities.ValidationUtils.ArgumentNotNull(Object value, String parameterName)
   at Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings)
   at RatTracking.ProgressDB.Init() in D:\Programmierung\Projects\RatScanner\RatScanner\RatTracking\ProgressDB.cs:line 23
   at RatScanner.RatScannerMain..ctor()
   at RatScanner.RatScannerMain.get_Instance()
   at RatScanner.View.BlazorUI.<>c.<.ctor>b__8_0(IServiceProvider s)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.CreateServiceAccessor(Type serviceType)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.GetService(Type serviceType)
   at Microsoft.AspNetCore.Components.ComponentFactory.<>c__DisplayClass7_0.<CreateInitializer>g__Initialize|1(IServiceProvider serviceProvider, IComponent component)
   at Microsoft.AspNetCore.Components.ComponentFactory.PerformPropertyInjection(IServiceProvider serviceProvider, IComponent instance)
   at Microsoft.AspNetCore.Components.ComponentFactory.InstantiateComponent(IServiceProvider serviceProvider, Type componentType)
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.InstantiateChildComponentOnFrame(RenderTreeFrame& frame, Int32 parentComponentId)
   at Microsoft.AspNetCore.Components.RenderTree.RenderTreeDiffBuilder.InitializeNewCom
```
</details>

**Comment by Blightbuster** [comment]
Could you please try again in a few minutes and check if the error persists?

**Comment by Blightbuster** [comment]
Thanks for reporting the issue, it has since been fixed in version 3.3.1.
You can launch rat scanner and it will ask you to update automatically.
